src/cat/cat.rs

```diff
@@ -301,6 +335,37 @@ fn write_fast(files: Vec<String>) -> CatResult<()> {
     }
 }
 
+/// This function is called from `write_fast()` on Linux and Android. The
+/// function `splice()` is used to move data between two file descriptors
+/// without copying between kernel- and userspace. This results in a large
+/// speedup.
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[inline]
+fn write_fast_using_splice(handle: &mut InputHandle, writer: RawFd) -> Result<(), nix::Error> {
+    const BUF_SIZE: usize = 1024 * 16;
+
+    let (pipe_rd, pipe_wr) = pipe()?;
+
+    loop {
+        let res = splice(
```

Comment:
I mentioned a possible way to add this to `uucore` rather than implementing it separately for each utility.  Let me know what you think.  I'm not opposed to merging this without the changes if you'd prefer, though.